File: Back End/Shop.Models/DB/Product.cs
using System.ComponentModel.DataAnnotations;

namespace Shop.Models.DB;

public class Product
{

	[Key]
	public int Id { get; set; }

	[Required]
	public required string Name { get; set; }

	[Required]
	public required float ManufacturePrice { get; set; }

	[Required]
	public required float WeightGrams { get; set; }
	[Required]
	public required float WeightMilliGrams { get; set; }

	[Required]
	public virtual required Group Group { get; set; }

	[Required]
	public required int GroupId { get; set; }


}

File: shop-front/src/app/interfaces/product.ts
import { GroupOutput } from './group';

export interface ProductInput {
  name: string;
  manufacturePrice: number;
  weightGrams: number;
  weightMilliGrams: number;
  groupId: number;
}

export interface ProductOutput {
  id: number;
  name: string;
  manufacturePrice: number;
  weightGrams: number;
  weightMilliGrams: number;
  group: GroupOutput;
}
